plugin/outputStream.go

```diff
@@ -41,22 +43,27 @@ func (s *OutputStream) Write(data []byte) (int, error) {
 	return len(data), err
 }
 
-// Close ensures the channel is closed when the last OutputStream is closed.
-func (s *OutputStream) Close() {
-	s.closer.Close()
-}
-
-// CloseWithError sends the given error before calling Close()
+// CloseWithError sends the given error before closing the OutputStream.
+// It will noop if the OutputStream's already closed.
 func (s *OutputStream) CloseWithError(err error) {
```

Comment:
I erased `Close` because I think it's simpler to expose a single method for closing an OutputStream. I also don't see a good reason why the OutputStream object should implement the `io.WriteCloser` interface when it just needs to implement `io.Writer` in order to do its job. Similar reasons explain why there is a `CloseStreamsWithError` method but not a `CloseStreams` method in `ExecCommandImpl`.

If there are good reasons for us to support the `Close` method in OutputStream, I'm happy to add it back.

Reply:
Also, exporting this is still a good idea in case the plugin API has disjoint Stdout/Stderr streams (e.g. like in the previous SSM Exec implementation). This way, plugin authors can still capture stream-specific failures.

Reply:
I can't think of a reason we need to keep the Closer functionality.